src/librustc/infer/error_reporting/nice_region_error/different_lifetimes.rs

```diff
@@ -99,30 +99,26 @@ impl<'a, 'gcx, 'tcx> NiceRegionError<'a, 'gcx, 'tcx> {
         let span_label_var1 = if let Some(simple_ident) = anon_arg_sup.pat.simple_ident() {
             format!(" from `{}`", simple_ident)
         } else {
-            format!("")
+            "".to_string()
```

Comment:
Using `String::new()` instead of `"".to_string()` would be even faster. (Similar for every other occurrences)